library/core/tests/slice.rs

```diff
@@ -2038,3 +2038,83 @@ fn test_slice_run_destructors() {
 
     assert_eq!(x.get(), 1);
 }
+
```

Comment:
In the previous PR, Lukas suggested some tests involving `usize::MAX`: https://github.com/rust-lang/rust/pull/62282#pullrequestreview-370747882

Can you add those?